src/biogeophys/HydrologyNoDrainageMod.F90

```diff
@@ -665,19 +666,35 @@ subroutine HydrologyNoDrainage(bounds, &
     ! Subroutines for storing the time derivative of top most soil layer moisture. This is
     ! used for diagnosing the downwards moisture flux within FAN.
 
-    subroutine store_tsl_moisture(waterstatebulk_inst)
+    subroutine store_tsl_moisture(waterstatebulk_inst, filter, num_fc)
       type(waterstatebulk_type), intent(inout) :: waterstatebulk_inst
-      associate(h2osoi_tend_tsl => waterstatebulk_inst%h2osoi_tend_tsl_col(bounds%begc:bounds%endc), &
-           h2osoi_liq_tsl => waterstatebulk_inst%h2osoi_liq_col(bounds%begc:bounds%endc,1))
-        h2osoi_tend_tsl = h2osoi_liq_tsl
-      end associate
+      integer, intent(in) :: filter(:)
+      integer, intent(in) :: num_fc
+
+      integer :: fc, c
+
+      do fc = 1, num_fc
+         c = filter(fc)
+         waterstatebulk_inst%h2osoi_tend_tsl_col(c) = waterstatebulk_inst%h2osoi_liq_col(c,1)
```

Comment:
I just noticed that you use `h2osoi_tend_tsl_col` for two purposes, the first of which is not a tendency at all. Please introduce an additional variable like `h2osoi_liq_col_saved` (which could probably be a subroutine-local variable in HydrologyNoDrainage) and use that one here, rather than using `h2osoi_tend_tsl_col` to save the old version of `h2osoi_liq_col`.